### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by MustakimMasum**
### Activities of this branch

- Applied SOLID principles to the main.rb file
- Refactored the process_option method to the app.rb file to keep the main.rb file only responsible to display the options to choose from and also exit from the app

Kindly asking for a review.

### CODE REVIEW

File: app.rb
Comment (Efiamotu-1) [inline app.rb:1]: - [ ] To follow the `Single Responsibility Rule` for best practices all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. see the example code snippet below
- code snippets
```ruby
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    person_type = gets.chomp

    print 'Title?'
    title = gets.chomp

    print 'Author?'
    author = gets.chomp
```
- [ ] Please kindly update your code base to follow this rule by implementing the above requirement everywhere in your app where you are requesting for user inputs :+1: :sparkles:

File: README.md [hunk 1]
```diff
@@ -82,6 +82,12 @@ No installation needed
 - GitHub: [@githubhandle](https://github.com/firepicaso)
 - LinkedIn: [LinkedIn](https://www.linkedin.com/in/mustakim-masum/)
 
+👤 **Ian Kiema**
+
+- GitHub: [@githubhandle](https://github.com/iankiema)
+- Twitter: [@twitterhandle](https://twitter.com/twitterhandle)
```
Comment (Efiamotu-1) [inline README.md]: - [ ] The link provided for the twitter handle is broken as the account doesn't exist, please kindly fix this broken link in your readme by providing the appropriate link to your twitter account :+1: :sparkles: 
- [ ] Also the link to your linkedIn account should also be provided for correctly :+1:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @firepicaso & @iankiema , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb
Comment (MustakimMasum) [reply app.rb:1]: Hi @Efiamotu-1, 

if you are available, can we get on a call, I have some confusion about this requirement.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @firepicaso and @iankiema,

✔️ Implementation of SOLID principles

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**